### DISCUSSION TITLE
How to focus on editor without closing the terminal

### DISCUSSION THREAD

**Discussion by volkan0x** [zed-industries/zed#10093] [Feature Requests]
I'd like to know how to focus on the editor while I am currently working on terminal, without closing the terminal?

Just like you can do with the "tab" key when you are searching with "cmd+f". You can focus on the editor with "tab" and come back to the search text write with "cmd+f" again with this:

>"context": "BufferSearchBar",
>    "bindings": {
>"tab": "buffer_search::FocusEditor",
>}

tried
>"context": "Terminal", {
>     "escape": ["terminal::FocusEditor"],
>}

to do that but couldn't make it.

**Comment by alygin** [comment]
I think, the actions you're looking for are `["workspace::ActivatePane", 0]` ... `["workspace::ActivatePane", 8]` with default key bindings `cmd-1` ... `cmd-9`.

**Comment by Moshyfawn** [comment]
I've got it to do exactly the opposite of the behaviour you're looking for, but the original keymap for this is ctrl-`, I think.

https://github.com/zed-industries/zed/assets/16290753/a738d1c9-63cb-491a-91c9-f45f99673eef

**Reply by volkan0x** [reply]
Not like this. Can you try pressing "tab" when you are on text search "cmd+f". You will immediately focus on editor with pressing tab while still searching a keyword. I want to do this on terminal panel with tab or escape key.

**Reply by mrnugget** [reply]
What about these?

```json
  {
    "context": "Dock",
    "bindings": {
      "ctrl-h": ["workspace::ActivatePaneInDirection", "Left"],
      "ctrl-l": ["workspace::ActivatePaneInDirection", "Right"],
      "ctrl-k": ["workspace::ActivatePaneInDirection", "Up"],
      "ctrl-j": ["workspace::ActivatePaneInDirection", "Down"]
    }
  }
```

**Reply by volkan0x** [reply]
It worked

**Comment by milosgajdos** [comment]
This has been driving me bonkers, too. What my brain does now is `CMD+J` and then back to 🥁 `CMD+J`

**Comment by mrnugget** [comment]
I still don't understand what the problem is: with `ctrl-backtick` you can change focus to and from terminal.

You can also overwrite it, here's the terminal part:

```json
  {
    //            👇 - Create binding for Terminal
    "context": "Terminal",
    "bindings": {
      //
      //                         👇 - On `esc` toggle back to editor
      "escape": "terminal_panel::ToggleFocus"
      //
      //
      //
    }
  }
```

That works for me.

https://github.com/zed-industries/zed/assets/1185253/774afa91-6a77-446e-aab3-8f77b711dcac

**Reply by milosgajdos** [reply]
Ooh, ok, so my problem was this:
* I didn't see this key shortcut in https://zed.dev/docs/key-bindings#terminal
* I found it in the UI menu, but it's not clear to me you can use it for switching b/w the pane and the terminal so I used to do  do combinations of CMD+J and ctrl+^

Thanks this is helpful.

**Reply by volkan0x** [reply]
Actually `ctrl-backtick` solves this problem. But I'd like to assign it to a custom shortcut. What is the definition of this on the bindings?

**Reply by mrnugget** [reply]
Exactly the thing that I posted above, in the JSON.

```json
 {
    "context": "Terminal",
    "bindings": {
      "escape": "terminal_panel::ToggleFocus"
    }
  }
```

**Reply by volkan0x** [reply]
It worked, thanks.
